From: Sergio Durigan Junior <sergiodj@sergiodj.net>
Date: Sun, 27 Dec 2015 01:15:54 -0500
Subject: Improve error message when "Trust this website" doesn't work

Sometimes the user will not be able to use the "Trust this website"
button, probably because she does not have gcr/gnome-keyring
installed.  This patch makes the warning/erro message displayed in
such cases more user-friendly.
---
 midori/midori-view.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/midori/midori-view.c b/midori/midori-view.c
index fbcb01c..00ea20a 100644
--- a/midori/midori-view.c
+++ b/midori/midori-view.c
@@ -623,7 +623,10 @@ midori_view_web_view_navigation_decision_cb (WebKitWebView*             web_view
                         gchar* slots = g_strjoinv (" , ", (gchar**)gcr_pkcs11_get_trust_lookup_uris ());
                         gchar* title = g_strdup_printf ("Error granting trust: %s", error->message);
                         midori_tab_stop_loading (MIDORI_TAB (view));
-                        midori_view_display_error (view, NULL, NULL, NULL, title, slots, NULL,
+                        midori_view_display_error (view, NULL, NULL, NULL, title, slots, _("Please look at our "
+                           "<a href=\"http://midori-browser.org/faqs/\" target=\"_blank\">FAQ</a>, section "
+                           "\"<a href=\"http://midori-browser.org/faqs/#security_features\" target=\"_blank\">"
+                           "Security Features</a>\", to understand how you can solve this problem."),
                             _("Trust this website"), NULL);
                         g_free (title);
                         g_free (slots);
